e2e/components/compare-widget/compare-widget.spec.js

```diff
@@ -0,0 +1,82 @@
+import compareWidgetComponent from './compare-widget.po';
+import SearchPage from '../../pages/search/search.po';
+import Hooks from '../../utilities/hooks';
+
+let component, hooks, page;
+let ListingId1 = 9261;
+let ListingId2 = 9956;
+let chplId2 = '15.04.04.2916.smar.07.01.1.190328';
+let chplId1 = '15.02.02.3007.A056.01.00.0.180214';
+
+beforeEach(async () => {
+    page = new SearchPage();
+    component = new compareWidgetComponent();
+    hooks = new Hooks();
+    await hooks.open('#/search');
+});
+
+describe('on compare widget', () => {
+    describe('if there is no listing added for compare', () => {
+
+        it('should not have compare products button', () => {
+            component.compareWidget.click();
+            assert.isFalse(component.compareProductsButton.isDisplayed());
+        });
+        it('should not have remove all products button', () => {
+            component.compareWidget.click();
+            assert.isFalse(component.removeProductsButton.isDisplayed());
+        });
+    });
+});
+
+describe('on compare widget', () => {
+    describe('if there is at least 1 listing added for compare', () => {
```

Comment:
```suggestion
    describe('if there is exactly 1 listing added for compare', () => {
```

If there were more than one, which the "at least" suggests, then some of these tests would fail

Reply:
Fixed